internal/driver/driver.go

```diff
@@ -277,6 +283,71 @@ func reportOptions(p *profile.Profile, vars variables) (*report.Options, error)
 	return ropt, nil
 }
 
+// identifyNumLabelUnits returns a map of numeric label keys to the units
+// associated with those keys.
+// Unit for a given key is the first encountered unit for that key. If multiple
+// units are encountered for values paired with a particular key, then the first
+// unit encountered is used and an error message is displayed.
+// If units are encountered for a particular key, the unit is then inferred
+// based on the key.
+func identifyNumLabelUnits(p *profile.Profile, ui plugin.UI) map[string]string {
```

Comment:
Looking more at this code, it seems that we should have this as a public method on Profile so that clients of pprof API can use it to get the mapping of tag keys to their units. We discussed before that we want to print the warnings / errors about the ignored units for the same tag key, but maybe we can do something like

```
func (p *Profile) NumLabelUnits() (map[string]string, map[string][]string) {
  ...
}
```

where the function returns the detected units per tag key and also the list of ignored duplicated units per tag key. The implementation here then would call that function and report the ignored units to the plugin UI.